platform/linux-generic/odp_packet.c

```diff
@@ -2146,7 +2146,7 @@ odp_packet_t odp_packet_ref(odp_packet_t pkt, uint32_t offset)
 	link_hdr->buf_hdr.seg[0].len  = seg->len  - seg_offset;
 	buffer_ref_inc(seg->hdr);
 
-	for (i = 1; i < num_copy; i++) {
+	for (i = 1; CONFIG_PACKET_SEGS_PER_HDR > 1 && i < num_copy; i++) {
```

Comment:
Not clear why this is necessary. If `CONFIG_PACKET_SEGS_PER_HDR == 1` then `num_copy` should be 1 here so the original `for` loop will also not execute. Or am I missing something here?

Reply:
Gcc throws an invalid 'array subscript is above array bounds' error if this extra clause is not added when CONFIG_PACKET_SEGS_PER_HDR=1.

Reply:
Which level of GCC? How about clang? I hate to mangle our code to accommodate a very specific compiler bug. If we need to so something like that at minimum it should be commented so that it can be cleaned up when the bug is eventually fixed.

Reply:
gcc 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.5). Clang doesn't complain about this line. I'll add a comment about the gcc bug.